### PR TITLE
[Data][Doc] Add doc to summarize different shuffle options

### LINKED ISSUE #41970: [data][docs] Improve user guides on randomly shuffling datasets
### Description

The current Ray Data docs don't cover many details on different ways to shuffle data.

### Link

_No response_

### PR DESCRIPTION
**Pull request by c21**
## Why are these changes needed?

Improve the section in documentation to summarize different shuffle options, so users have a place to understand what shuffle options we have.

## Related issue number

Closes https://github.com/ray-project/ray/issues/41970

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/data/transforming-data.rst [hunk 1]
```diff
@@ -305,12 +305,64 @@ To transform groups, call :meth:`~ray.data.Dataset.groupby` to group rows. Then,
...
+    import ray
+
+    ds = ray.data.read_images(
+        "s3://anonymous@ray-example-data/image-datasets/simple",
+        shuffle="files",
+    )
+
+.. tip::
+
+    This is the fastest option for shuffle, and is purely metadata operation. This
```
Comment (bveeramani) [inline doc/source/data/transforming-data.rst:331]: I feel like a word is missing here?
```suggestion
    This is the fastest option for shuffle, and is a purely metadata operation. This
```

File: doc/source/data/transforming-data.rst [hunk 2]
```diff
@@ -305,12 +305,64 @@ To transform groups, call :meth:`~ray.data.Dataset.groupby` to group rows. Then,
...
+    option does not shuffle the actual rows inside files, so the randomness would be
```
Comment (bveeramani) [inline doc/source/data/transforming-data.rst:332]: https://developers.google.com/style/contractions#:~:text=In%20particular%2C%20it's%20fine%20to,misread%20don't%20as%20do.
```suggestion
    option doesn't shuffle the actual rows inside files, so the randomness might be
```

File: doc/source/data/transforming-data.rst [hunk 3]
```diff
@@ -323,8 +375,8 @@ To randomly shuffle all rows, call :meth:`~ray.data.Dataset.random_shuffle`.
 
 .. tip::
 
-    :meth:`~ray.data.Dataset.random_shuffle` is slow. For better performance, try
-    :ref:`Iterating over batches with shuffling <iterating-over-batches-with-shuffling>`.
+    This is the slowest option for shuffle, and requires tranasferring data across
```
Comment (bveeramani) [inline doc/source/data/transforming-data.rst:378]:
```suggestion
    This is the slowest option for shuffle, and requires transferring data across
```

**Review by bveeramani — APPROVED** [review]
Nice

File: doc/source/data/transforming-data.rst (hunk 1 above)
Comment (c21) [reply doc/source/data/transforming-data.rst]: updated.

File: doc/source/data/transforming-data.rst (hunk 2 above)
Comment (c21) [reply doc/source/data/transforming-data.rst]: updated.

File: doc/source/data/transforming-data.rst (hunk 3 above)
Comment (c21) [reply doc/source/data/transforming-data.rst]: my bad, updated.

**Pull request merged**